### PR TITLE
Toast message improvement

### LINKED ISSUE #1039: Toast messages improvement
- Have timer progress bar under toast messages (timer increase to 4 seconds)
- Make toast messages slightly higher and add dismissable X button

### PR DESCRIPTION
**Pull request by MTSaeed**
**What does this PR?**

Adds progress bar under the toast message. The color of progress bar is light green , red2 and blue for success , error and info toaster respectively.
Adds the close button in toaster container to close the toaster.
When mouse enters in the feedback container, it stops the progress bar until mouse leaves from the progress bar.
When mouse leaves from the progress bar, it starts the progress bar from where he left it.
You can get a better idea from the  attached video.

**Steps to reproduce**

1. Copy any message and hover on the feedback container. You will notice, the progress bar stops. When Mouse leaves from the progress bar, the progress starts its working from where he left.
2. You can follow the steps to test it by following the steps that i did in the video.

**Issue number** 

fixes #1039 

**Screenshot / Video**

https://user-images.githubusercontent.com/106739598/225619074-d1fddda5-540a-401e-be26-a37ea61b024c.mp4

### CODE REVIEW

**Comment by MTSaeed** [comment]
image

**Comment by MTSaeed** [comment]
Hy @feruzm .

I have made some changes in the styling of the toaster. You can compare these changes with the existing toaster of ecency. I hope you would like these changes. Kindly let me know if you like these changes and i will push the updated code with changes.
Thanks!

https://user-images.githubusercontent.com/106739598/225651854-0311d545-eddb-490b-abc4-3586c654e684.mp4

**Comment by feruzm** [comment]
@MTSaeed In last video, night theme doesn't look good.

**Review by feruzm — APPROVED** [review]
@MTSaeed LGTM, please add test to feedback component, after that we can merge this as well. I did improve colouring a bit.

File: src/common/components/feedback/index.tsx [hunk 1]
```diff
@@ -60,26 +60,80 @@ interface State {
...
+      if (this.state.intervalState) {
+        this.setState({ intervalState: false });
+        return;
+      }
+      const { progress } = this.state;
+      this.stateSet({ progress: progress - 2.5 });
+    };
+
+    const interval = setInterval(
+      (function hello() {
```
Comment (dkildar) [inline src/common/components/feedback/index.tsx:113]: What is `hello`?:) Please specify name.

File: src/common/components/feedback/index.tsx [hunk 2]
```diff
@@ -89,74 +143,131 @@ export default class Feedback extends BaseComponent<Props, State> {
...
+                            ) : (
+                              <></>
+                            )}
+                            {!ErrorTypes.INFO && (
+                              <Button
+                                className="mt-2 details-button px-0"
+                                variant="link"
+                                onClick={() =>
+                                  window.open(
+                                    "mailto:bug@ecency.com?Subject=Reporting issue&Body=Hello team, \n I would like to report issue: \n",
```
Comment (dkildar) [inline src/common/components/feedback/index.tsx:227]: I guess these sentences should be translated

File: src/common/components/feedback/index.tsx [hunk 3]
```diff
@@ -60,26 +60,80 @@ interface State {
   list: FeedbackObject[];
   showDialog: boolean;
   detailedObject: FeedbackObject | null;
+  display: boolean;
+  progress: number;
+  intervalState: boolean;
 }
 
 export default class Feedback extends BaseComponent<Props, State> {
```
[the remark this answers is not in the document]
Comment (dkildar) [reply src/common/components/feedback/index.tsx:68]: And second one: after moving you can use `useInterval` hook from `react-use` for timer declaration

File: src/common/components/feedback/index.tsx (hunk 2 above)
Comment (feruzm) [reply src/common/components/feedback/index.tsx:227]: https://github.com/ecency/ecency-vision/pull/1196#discussion_r1145387045

File: src/common/components/feedback/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (feruzm) [reply src/common/components/feedback/index.tsx:68]: @MTSaeed don't forget to create test for component.

File: src/common/components/feedback-message/index.tsx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    const itemExists = list.find((x) => x.message === props.feedback.message);
+    if (itemExists) {
+      return;
+    }
+    setList([...list, props.feedback]);
+    startTimer();
+  }, []);
+
+  useEffect(() => {
+    console.log(list);
```
Comment (dkildar) [inline src/common/components/feedback-message/index.tsx:45]: Are you need this?

**Review by feruzm — APPROVED** [review]
lgtm

**Pull request merged**